components/gly_gen/index.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import { MetaNode } from '@/spec/metanode'
+import { GeneInfo } from '../service/mygeneinfo'
+import { z } from 'zod'
+
+export const GlyGenGeneType = {
+  gene_name: typeof GeneInfo
+}
+
+export const GlyGenResponse = z.object({
+  queryinfo: z.any(),
+  results: z.array(z.any())
+})
+
+export type GlyGenResponseType = z.infer<typeof GlyGenResponse>
+
+export const GlyGenResponseNode = MetaNode.createData('GlyGenResponse')
+  .meta({
+    label: 'GlyGen Response',
+    description: 'GlyGen response object',
+  })
+  .codec(GlyGenResponse)
+  .view(data => (
+    <>
+      <pre>{JSON.stringify(data.results)}</pre>
+    </>
+    
+  ))
+  .build()
+
+export const ProteinProductInformation = MetaNode.createProcess('ProteinProductInformation')
+  .meta({
+    label: 'Protein Product Information',
+    description: 'Search for protein records in GlyGen',
+  })
+  .inputs({ gene: GeneInfo })
+  .output(GlyGenResponseNode)
+  .resolve(async (props) => {
+    const query = encodeURIComponent(JSON.stringify({
+      recommended_gene_name: props.inputs.gene.symbol,
+    }))
+    const request = await fetch(`https://api.glygen.org/directsearch/protein/?query=${query}`, {
+      method: 'GET',
+      headers: {
+        accept: 'application/json'
+      }
+      })
+    const response = await request.json()
+  })
```

Comment:
It looks like you forgot to return the response in the `resolve` function, that causes an error elsewhere which expects json to come out of this method.

```tsx
return response
```

Reply:
That is embarrassing. Thanks for your patience!